### ISSUE TITLE
can't connect to server

### ISSUE THREAD

**Issue by Jadan1213** [Viola-Siemens/Mod-Whitelist#4]
MC: 1.20.1
Forge: 47.2.17
Mod whitelist: 1.1.0

I installed mod whitelist on server and client, but it won't allow connection, even when the whitelist is currently empty (aside from "forge") as i haven't configured the whitelist yet. 'Only allow whitelist' is set to "false".

When trying to connect, all it says to the client is failed to connect to server. And the server console doesn't even show a connection attempt

![image](https://github.com/Viola-Siemens/Mod-Whitelist/assets/68805162/88e33020-1e52-4dca-a30d-8bc4a9ae962f)

**Comment by Viola-Siemens** [comment]
If there's no connection attempt shown in the console, it may cause by the firewall, or incorrect ip address and port inputted on your client side. Furthermore, check whether you logged in with online mode or not.

**Comment by Jadan1213** [comment]
> If there's no connection attempt shown in the console, it may cause by the firewall, or incorrect ip address and port inputted on your client side. Furthermore, check whether you logged in with online mode or not.

Firewall is not blocking it. 
Server address is correct. 

Disabling the mod on both sides allows connection. 

Does the mod use udp traffic? I'm using TCP shield, and I believe only routes TCP traffic on the free plan.

**Comment by Viola-Siemens** [comment]
> Does the mod use udp traffic? I'm using TCP shield, and I believe only routes TCP traffic on the free plan.

No, only adds some more extra carries to the login package. This may caused by no ModWhitelist mod or mismatched version on client side.

**Comment by Jadan1213** [comment]
Both client and server versions are 1.1.0.

Connecting locally, does work, however the firewall is NOT blocking connections, as without mod whitelist, there is zero issues connecting to the server externally.

My best guess is something to do with the connection going through the TCP Shield service.

**Comment by Jadan1213** [comment]
i'm going to set up an alternate dns record, and attempt to connect to the server, bypassing tcp shield to see if it connects without it.

**Comment by Jadan1213** [comment]
bypassing tcp shield and connecting to the server with mod whitelist enabled does work. I'm going to open a ticket with them. maybe they will know what's up.

**Comment by Viola-Siemens** [comment]
I think maybe tcp shield think the package is too big for logging in so it refuse your connection if you install a number of mods with out Mod Whitelist. Thanks for letting me know about it.

**Comment by Jadan1213** [comment]
my client modpack is 384 mods (fully loaded it shows around 450? thanks to sinytra and all the mini translation mods it uses)

**Comment by Jadan1213** [comment]
I don't know why it wouldn't allow the connection tbh, because it's literally just a one mod difference lol. 

I don't know how much data is added to the payload with mod whitelist so... lol

**Comment by Jadan1213** [comment]
If you manage to come up with a workaround, i'd be more than happy to be a beta tester for it.

**Comment by Jadan1213** [comment]
> I think maybe tcp shield think the package is too big for logging in so it refuse your connection if you install a number of mods with out Mod Whitelist. Thanks for letting me know about it.

also, this is being used as merely a proxy, there's no additional mods/packages installed on the server in order to use it.
